### ISSUE TITLE
ES6 Intellisense issues

### ISSUE THREAD

**Issue by nrgjack** [Microsoft/nodejstools#674]
Hi everyone, 

i have vs2015 enterprise + sp1 + ntvs latest version (1.1)

i'm wondering to switch from full intellisense to es6 intellisense

i made some test with a file called mymodule.js
```javascript
'use strict';

var testFunction = function () {
    return true;
};

var testObject = {
    item: 1
};

module.exports = {
    version: '1.0.0',
    testFunction: testFunction,
    testObject: testObject
};
```
and a file called starter.js
```javascript
'use strict';

var _ = require('lodash');
var m = require('./mymodule');

// TODO : test here
```

## issue 1)
i noticed List Members do not filter above the items belonging to the object i'm working with, 
for example lodash.<CTRL+SPACE> shows members belonging to my custom modules

![image](https://cloud.githubusercontent.com/assets/11472052/12673903/09fc3e60-c680-11e5-8a1c-d539c480ca2f.png)
and viceversa
![image](https://cloud.githubusercontent.com/assets/11472052/12673909/17dfe248-c680-11e5-9ffd-13eb75a2217a.png)

## issue 2)
Parameter Info is not working with my module's functions
![image](https://cloud.githubusercontent.com/assets/11472052/12673920/2a63213c-c680-11e5-8f4b-b9f679614a65.png)

**switching back to full intellisense everything is back to normal **
![image](https://cloud.githubusercontent.com/assets/11472052/12673783/18a733e4-c67f-11e5-93dc-7be7b50012c0.png)

![image](https://cloud.githubusercontent.com/assets/11472052/12673769/02830a84-c67f-11e5-9153-91f0d60d4a70.png)

could you tell me if i misconfigured something or it is a real issue ? 

thank you guys

**Comment by beautifulcoder** [comment]
Awesome findings. I am experience issues when I turn on ES6 intellisense. It is not clear that TypeScript editor options apply when you turn this on. Tabbing was all over place, I changed it in node but not TypeScript. The labels denoting browser and node code in solution explorer don't show.

**Comment by mjbvz** [comment]
Thank you for reporting these findings.

Since this issue originally open, the NTVS team, as well as the Typescript team that powers our modern IntelliSense experience, have both been working hard to improve the Node.js IntelliSense in VS. We just released [NTVS 1.2 beta](https://github.com/Microsoft/nodejstools/releases), and I recommend downloading it and trying it out. There's still work to be done, including an issue with IntelliSense for local files (#1076) that requires VS2015 Update 3 or a manual work arount, but the overall experience is much improved over NTVS 1.1

For the first issue you noted, we now use `*.d.ts` typings file to provide rich and powerful IntelliSense for many common node modules. This should address many of the problems you saw in the completion list:
![image](https://cloud.githubusercontent.com/assets/12821956/16251205/0358089e-37d9-11e6-9351-9c29067bd364.png)

The second issue, IntelliSense for local modules, is still broken in the currently available VS2015 release, but should be fixed with VS2015 Update 3 RTM when using NTVS 1.2 Beta. The issue is being tracked by #1076, and there's a workaround you can try to get IntelliSense working properly on VS2015 Update 2.


Please give the new beta a try and report any new issues you run into. Thanks
